### PR TITLE
docs: Improve Vector README to clarify purpose and position

### PR DESCRIPTION
**Pull request by binarylogic**
In light of recent discussions, I wanted to clarify Vector's purpose and market position. Feedback and changes are welcome!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,25 +113,57 @@ fundamental differences in quality and design:
...
-## The Little Details Make All Of The Difference
+## The Little Details
 
-* **Zero delay start** - Starts and restarts without a delay.
+#### Data Model
+
+* **All data types** - [Logs][docs.data-model.log], [metrics][docs.data-model.metric], and traces (coming soon).
+* **Customizable log schema** - [Change Vector's log schema][docs.global-options#log_schema] to anything you like.
+* **Rich type system** - Support for [JSON primitive types][docs.data-model.log#types] and [timestamps][docs.data-model.log#timestamps].
+* **Metrics interoperability** - A [sophisticated metrics data model][docs.data-model.metric] ensures _correct_ interoperality between systems.
```
Comment (lukesteensen) [inline README.md]: Not sure if "interoperality" is a word or not. Maybe just "integration"?

File: README.md [hunk 2]
```diff
@@ -96,25 +113,57 @@ fundamental differences in quality and design:
...
+* **Metrics aggregation** - Aggregated [histgorams][docs.data-model.metric#aggregated_histogram] and [summaries][docs.data-model.metric#aggregated_summary] allow for aggreation without loss of precision.
```
Comment (lukesteensen) [inline README.md]: Typo: "aggreation"

Maybe also swap out "allow for aggregation" (redundant) with something like "reduce cost/volume".

File: README.md [hunk 3]
```diff
@@ -96,25 +113,57 @@ fundamental differences in quality and design:
...
+
+#### Data Flow
+
+* **Pipelining** - A [directed acyclic graph processing model][docs.configuration] allows for flexible topologies.
+* **Control-flow** - Transforms like the [`swimlanes` transform][docs.transforms.swimlanes] allow for complex control-flow logic.
+* **Dynamic partitioning** - Create dynamic partitions on the fly ([ex][docs.sinks.aws_s3#partitioning]) with Vector's [templating syntax][docs.reference.templating].
+
+#### Data Processing
+
+* **Multi-line merging** - [Merge multi-line logs][docs.sources.file#multiline] into one event, such as stacktraces.
```
Comment (lukesteensen) [inline README.md]: I'd probably move this down the list a bit. The others are more interesting, core features.

File: README.md [hunk 4]
```diff
@@ -96,25 +113,57 @@ fundamental differences in quality and design:
...
+
 * **Decoupled buffer design** - Buffers are coupled with sinks; a bad sink won't bring the entire pipeline to a halt.
-* **Thoughtful disk buffering** - Transfers data to operating system memory to reduce the risk of data loss.
-* **Merge split logs** - Easily merge multi-line logs into one event.
-* **Custom DNS** - [Custom DNS][docs.global-options#dns_servers] makes service discovery possible.
-* **Rate-limited logging** - Vector's internal logging is rate-limited and noise-free.
-* **Optionally fully static binary** - Optional MUSL static binaries mean zero required dependencies.
-* **Thoughtful docs** - [Quality documentation][docs.what-is-vector] that respects your time and reduces communication overhead.
-* **Clear Guarantees** - A [guarantee support matrix][docs.guarantees] helps you make the appropriate tradeoffs.
+* **Fibnacci backoffs with jitter** - Vector's backoff algorithm makes Vector a good citizen during outages.
```
Comment (lukesteensen) [inline README.md]: Typo: "Fibnacci"

Might be easier to just say "Intelligent retries" or something like that.

File: README.md [hunk 5]
```diff
@@ -96,25 +113,57 @@ fundamental differences in quality and design:
...
+* **Back pressure & load shedding** - Buffers can be configured to provide back pressure or shed load.
```
Comment (lukesteensen) [inline README.md]: I usually see backpressure written as one word in technical contexts.

File: README.md [hunk 6]
```diff
@@ -96,25 +113,57 @@ fundamental differences in quality and design:
...
+* **Memory safety** - Vector is built in Rust and is memory safe, avoiding a large class of memory related errors.
+* **Rate-limited internal logging** - Vector's internal logging is rate-limited avoiding IO saturation when errors occur.
+* **Sink healthchecks** - Healthchecks provide startup safety and prevent deploys with bad configuration.
+* **Durable disk buffering** - Vector uses `leveldb` which transfers data to OSm memory for improved durability.
```
Comment (lukesteensen) [inline README.md]: I don't know what "transfers data to OSm memory" means. Maybe something like: Vector uses `leveldb` for robust data durability across restarts.

File: README.md (hunk 1 above)
Comment (binarylogic) [reply README.md]: Yeah, that's a typo. I meant "interoperability".

**Comment by binarylogic** [comment]
Thanks for the review! I addressed all points.

**Pull request merged**